src/main/java/org/broad/igv/sam/AlignmentDataManager.java

```diff
@@ -548,21 +553,35 @@ public void alleleThresholdChanged() {
     }
 
     public boolean isTenX() {
-        return reader.isTenX();
+        return checkReader().isTenX();
     }
 
     public boolean isPhased() {
-        return reader.isPhased();
+        return checkReader().isPhased();
     }
 
     public boolean isMoleculo() {
-        return reader.isMoleculo();
+        return checkReader().isMoleculo();
     }
 
     public Collection<AlignmentInterval> getLoadedIntervals() {
         return intervalCache;
     }
 
+    private AlignmentTileLoader checkReader() {
```

Comment:
Why is this necessary (creating a new reader for every access t reader)?

Reply:
We for sure don't want to be recreating an alignment tile loader.

Reply:
This is needed because of the nature of time-gated access to resources on AWS (using pre-signed urls) and it only affects performance (if that was the concern) if we are handling an AWS path/url:

```java
    private AlignmentTileLoader checkReader() {
        try {
            String aPath = locator.getPath();
            if (AmazonUtils.isAwsS3Path(aPath) && !AmazonUtils.isS3PresignedValid(aPath)) {
                reader = new AlignmentTileLoader(AlignmentReaderFactory.getReader(locator));
            }
        } catch(MalformedURLException e){
            e.printStackTrace();
        } catch (IOException e) {
            e.printStackTrace();
        }

        return reader;
    }
```

If this is not done, pre-signed urls expires after 1h (maximum IAM role time duration) and the user cannot access the resource anymore (unless he/she logs out and logs in again manually).

Reply:
Understand the problem, I don't think this is the right solution.

Reply:
If it were easier to type I would expound.   I  will look more deeply tomorrow.  Getting late here.

Reply:
A quick change that would be an improvement would be to just update te reader property of the alignmenttileloader, rather than create a new a-t-l.   But ill look tomorrow for better solutions

Reply:
Absolutely! We (@reisingerf and I) are more than happy to hear about better options that work with AWS.

This is the only way we could find for this to work with minimal changes to the existing codebase (no extra classes, complexity, etc...).

Reply:
Yes, understood,  not a criticism.  I will work on this.

Reply:
Hi @jrobinso , you can blame this one mainly on me and my general ignorance about the code base.

As @brainstorm mentioned, it's what we thought best at the time, not having the most in-depth knowledge and understanding of the code base. To give you an idea about our thought process:

The AlignmentTileLoader looked to us as a light-weight wrapper around the AlignmentReader and did not seem to add a lot of instantiation overhead. On the other hand the reader property is a private property and we did not want to run the risk of compromising/corrupting any AlignmentReader - AlignmentTileLoader relationships. Also a new AlignmentTileLoader is only created if/when a pre-signed URL is accessed after it had expired; to us an insignificant overhead that does not happen very frequently.

Therefore, it seemed cleaner and saver to keep that part of the data access management in the AlignmentDataManager. Having said that, we'd be more than happy to hear about better approaches.

Reply:
@reisingerf thanks for the explanation, yes absolutely the safest way to do this.  And its acceptable as a solution.  

I would like to explore doing this url refresh at a much lower level, perhaps in HttpUtils.  What happens for example when an AWS bigwig file url expires?   Actually it looks like it is taken care of there, why isnt this sufficient  (HttpUtils?

```
   public static String mapURL(String urlString) throws MalformedURLException {

        if (urlString.startsWith("gs://")) {
            urlString = GoogleUtils.translateGoogleCloudURL(urlString);
        } else if (AmazonUtils.isAwsS3Path(urlString)) {
            try {
                urlString = AmazonUtils.translateAmazonCloudURL(urlString);
  
    }
```

Reply:
That's what we were hoping as well. Unfortunately the reader is further downstream compared to the HttpUtils.
Basically the S3 URL (s3://bucket/object.key) of the ResourceLocator is translated into a pre-signed URL (https://...), via HttpUtils, before creating the reader. The reader is bound to and caches that pre-signed URL, which will eventually expire.
There is, to our knowledge, no way to update the URL/source a reader is configured with. Either a reader does not support the update of the source URL (as is common in readers, see Java FileReader) or the reader could be updated, but too far downstream were code could either not be changed or references to the original ResourceLocator were long lost and only the translated URL was passed on.

In the end we thought the only consistent and clean way is to replace the entire reader (which makes sense, but is unfortunate for the use case of pre-signed URLs).

Reply:
OK, so this is a design issue out of scope for this PR.  We will go with your solution and I'll take on the larger issue separately.  Ultimately I think we want to keep the original s3 urls in ResourceLocator and treat the presigned URLs as transient entities generated as needed.  If I had both hands at my disposal I would just do that now :).

Reply:
@jrobinso Neat idea, but not sure that would work. 
First up it already breaks in HttpUtil.mapURL, as s3:// is not a supported URL protocol. Even if that would be fixed (and I don't know the implications of translating the S3 URL in multiple places) I don't know if all readers that cache the URL would fall back onto those HttpUtil methods (happy to test, if we get to a stage where all previous errors are resolved).
At this point why not delay those changes for a PR down the road? 
Our proposed changes should only affect AWS S3 URLs, so the only risk is that not all formats/readers are supporting AWS resources. Granted, it's not nice, but nothing that can't be fixed down the road.

Reply:
Yep, `aws-support-test` breaks with:

```java
Error loading BAM file: java.net.MalformedURLException: unknown protocol: s3
```

When trying to load pretty much any resource from S3 (since `s3://` is not a `URL` as per RFC2396).